src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPropertySymbol.cs

```diff
@@ -7,338 +7,143 @@
-        public override MethodSymbol GetMethod { get; }
-        public override MethodSymbol SetMethod { get; }
-        public override NamedTypeSymbol ContainingType { get; }
+        public ParameterSymbol BackingParameter { get; }
 
         public SynthesizedRecordPropertySymbol(
-            NamedTypeSymbol containingType,
+            SourceMemberContainerTypeSymbol containingType,
+            Binder binder,
             ParameterSymbol backingParameter,
             DiagnosticBag diagnostics)
-            : base(backingParameter.Locations[0])
+            : base(containingType,
+                binder,
+                backingParameter.GetNonNullSyntaxNode(),
+                backingParameter.Name,
+                backingParameter.Locations[0],
+                diagnostics)
         {
-            ContainingType = containingType;
-            _backingParameter = backingParameter;
-            string name = backingParameter.Name;
-            BackingField = new SynthesizedBackingFieldSymbol(
-                this,
-                GeneratedNames.MakeBackingFieldName(name),
-                isReadOnly: true,
-                isStatic: false,
-                hasInitializer: true);
-            GetMethod = new GetAccessorSymbol(this, name);
-            SetMethod = new InitAccessorSymbol(this, name, diagnostics);
+            BackingParameter = backingParameter;
         }
 
-        public ParameterSymbol BackingParameter => _backingParameter;
-
-        internal override bool IsAutoProperty => true;
-
-        public override RefKind RefKind => RefKind.None;
-
-        public override TypeWithAnnotations TypeWithAnnotations => _backingParameter.TypeWithAnnotations;
-
-        public override ImmutableArray<CustomModifier> RefCustomModifiers => ImmutableArray<CustomModifier>.Empty;
-
-        public override ImmutableArray<ParameterSymbol> Parameters => ImmutableArray<ParameterSymbol>.Empty;
-
-        public override bool IsIndexer => false;
-
-        public override ImmutableArray<PropertySymbol> ExplicitInterfaceImplementations => ImmutableArray<PropertySymbol>.Empty;
-
-        public override Symbol ContainingSymbol => ContainingType;
-
-        public override ImmutableArray<Location> Locations => _backingParameter.Locations;
-
-        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => _backingParameter.DeclaringSyntaxReferences;
-
-        public override Accessibility DeclaredAccessibility => Accessibility.Public;
-
-        public override bool IsStatic => false;
-
-        public override bool IsVirtual => false;
-
-        public override bool IsOverride => false;
+        protected override TypeSyntax GetTypeSyntax(SyntaxNode syntax)
+            => ((ParameterSyntax)syntax).Type!;
 
-        public override bool IsAbstract => false;
+        protected override SyntaxTokenList GetModifierTokens(SyntaxNode syntax)
+            => ((ParameterSyntax)syntax).Modifiers;
```

Comment:
>((ParameterSyntax)syntax).Modifiers; [](start = 15, length = 36)

This doesn't feel right. What parameter modifiers have to do with property modifiers?